packages/stripe-integration/src/stripe-upe/stripe-upe-integration-service.spec.ts

```diff
@@ -0,0 +1,667 @@
+        });
+
+        it('skip subscription actions if no stripe payment element found', () => {
+            stripeElementsMock.getElement = () => null;
+
+            stripeUPEIntegrationService.initCheckoutEventsSubscription(
+                gatewayId,
+                methodId,
+                stripeupeMock,
+                stripeElementsMock,
+            );
+
+            expect(paymentIntegrationService.loadPaymentMethod).not.toHaveBeenCalled();
+        });
+
+        it('throws error if loadPaymentMethod fails', async () => {
+            stripeupeMock.onError = jest.fn();
+            jest.spyOn(paymentIntegrationService, 'loadPaymentMethod').mockRejectedValue(
+                new Error(),
+            );
+
+            stripeUPEIntegrationService.initCheckoutEventsSubscription(
+                gatewayId,
+                methodId,
+                stripeupeMock,
+                stripeElementsMock,
+            );
+
+            await new Promise((resolve) => process.nextTick(resolve));
+
+            expect(paymentIntegrationService.loadPaymentMethod).toHaveBeenCalled();
+            expect(stripeupeMock.onError).toHaveBeenCalled();
+            expect(stripeElementMock.unmount).not.toHaveBeenCalled();
+        });
+
+        it('unmount stripe payment element if loadPaymentMethod fails', async () => {
+            jest.spyOn(paymentIntegrationService, 'loadPaymentMethod').mockRejectedValue(
+                new Error(),
+            );
+            jest.spyOn(document, 'getElementById').mockReturnValue(document.createElement('div'));
+
+            stripeUPEIntegrationService.mountElement(stripeElementMock, stripeupeMock.containerId);
+            stripeUPEIntegrationService.initCheckoutEventsSubscription(
+                gatewayId,
+                methodId,
+                stripeupeMock,
+                stripeElementsMock,
+            );
+
+            await new Promise((resolve) => process.nextTick(resolve));
+
+            expect(stripeElementMock.unmount).toHaveBeenCalled();
+            // eslint-disable-next-line @typescript-eslint/dot-notation, dot-notation
+            expect(stripeUPEIntegrationService['isMounted']).toBe(false);
+        });
+
+        it('mount stripe payment element if not mounted', async () => {
+            jest.spyOn(document, 'getElementById').mockReturnValue(document.createElement('div'));
+
+            stripeUPEIntegrationService.initCheckoutEventsSubscription(
+                gatewayId,
+                methodId,
+                stripeupeMock,
+                stripeElementsMock,
+            );
+
+            await new Promise((resolve) => process.nextTick(resolve));
+
+            expect(stripeElementsMock.fetchUpdates).toHaveBeenCalled();
+            expect(stripeElementMock.mount).toHaveBeenCalled();
+            // eslint-disable-next-line @typescript-eslint/dot-notation, dot-notation
+            expect(stripeUPEIntegrationService['isMounted']).toBe(true);
```

Comment:
Why don't you use dot notation here?

Reply:
To check value of private variable. To make this check more clear and don't mix tests from different methods of this class.

Reply:
It is not the good practice, but in current case I think will be better to check only one variable without additional complex logic with multiple calls of different methods.